cpp/include/kvikio/file_handle.hpp

```diff
@@ -62,25 +63,43 @@ inline int open_fd_parse_flags(const std::string& flags)
     default: throw std::invalid_argument("Unknown file open flag");
   }
   file_flags |= O_CLOEXEC;
-  if (!defaults::compat_mode()) { file_flags |= O_DIRECT; }
+  if (o_direct) { file_flags |= O_DIRECT; }
   return file_flags;
 }
 
 /**
  * @brief Open file using `open(2)`
  *
  * @param flags Open flags given as a string
+ * @param o_direct Append O_DIRECT to `flags`
  * @param mode Access modes
  * @return File descriptor
  */
-inline int open_fd(const std::string& file_path, const std::string& flags, mode_t mode)
+inline int open_fd(const std::string& file_path,
+                   const std::string& flags,
+                   bool o_direct,
+                   mode_t mode)
```

Comment:
Is this a public API or intended for internal usage? Is this a breaking API change? Should we be adding this argument in the middle of existing ones? Should it have a default?

Reply:
No, this is private. In a followup PR I will begin using `namespace kvikio::detail` for internal functions like this one.